main.js

```diff
@@ -184,18 +186,23 @@ ipcMain.on('error', (e, errorObj) => {
 // uninstall addon listener
 ipcMain.on('uninstallAddon', (e, addonObj) => {
   console.log(`Received request to delete ${addonObj.name}`)
-  installedAddonsDict = uninstallAddon(addonObj, configObj, installedAddonsDict)
-  saveToAddonList(configObj, installedAddonsDict)
-  console.log(addonObj.name)
-  mainWindow.webContents.send('modAddonObj', {
-    'displayName': addonObj.displayName,
-    'name': addonObj.name,
-    'version': addonObj.version,
-    'host': addonObj.host,
-    'URL': addonObj.URL,
-    'authors': addonObj.authors,
-    'status': 'NOT_INSTALLED'
-  })
+  try {
```

Comment:
This seems like a Java approach to running these statements, I'm not too sure if I think this is a good design pattern. What do you think about sticking to the promise then/catch statements?

Reply:
rewritten